### DISCUSSION TITLE
Merge package.json root to package/app package.json level

### DISCUSSION THREAD

**Discussion by Jony-Y** [nrwl/nx#16328] [Feature Requests]
We are building a new NX-based Monorepo, and while I read and understood the capabilities that NX provides over package.json. I would still want to have the ability per project to override the global dependencies with my own. Similar to the NPM decision tree (The one closest to the root of the project will win).

I couldn't see if this is automatically implemented in NX and if not, would I have to write my own executor that will do that merge between package.json files?

**Comment by kevindice** [comment]
I believe more or less Nx enforces a one-version-policy by default by having you symlink each package's node_modules to the repo-root node_modules.

That said, it looks like you can utilize yarn workspaces, and this gives you the ability to have common dependencies hoisted up to the top-level `node_modules` and differing differing dependency versions not hoisted (i.e. located in each module's own node_modules folder) 

If possible though, a one-version-policy is a nice thing to adopt.  It's good for bundle sizes if you're building for a platform where that's important.
